compression_handler.go

```diff
@@ -10,6 +10,7 @@ import (
 	abstractions "github.com/microsoft/kiota-abstractions-go"
 	"go.opentelemetry.io/otel"
 	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.24.0"

```

Comment:
```suggestion
	semconv "go.opentelemetry.io/otel/semconv"
```
Should the imports include the version here? Or will they cause issue incase of an update in the go.mod file?

Reply:
It did not work when I did not assign the version here. So if you manage to get it working that'd be great :)

Reply:
@andrueastman you need to reference the version and yes, it will cause runtime errors if you upgrade the otel sdk and don't have the correct version of `semconv` referenced. I hace seen multiple issues and proposals for this but no progress I'm aware of to mitigate having to update all references of `semconv` usage. See https://github.com/open-telemetry/opentelemetry-go/issues/4886

Reply:
In general we've faced similar issues in other languages. This dependency causes a lot of issues since they ship breaking changes without major revs (to keep the version aligned with the other packages roughly). We've resulted NOT to use this package in other languages because of that. Yes it introduces misalignment possibilities/added maintenance burden because of that.
I hope this additional context helps!